src/solver/infeasible-problem-analysis/problem.h

```diff
@@ -11,22 +11,77 @@ namespace Antares
 {
 namespace Optimization
 {
-class InfeasibleProblemAnalysis
+
+
+class SingleAnalysis
 {
 public:
-    InfeasibleProblemAnalysis() = delete;
-    explicit InfeasibleProblemAnalysis(const std::string& solverName, const PROBLEME_SIMPLEXE_NOMME* ProbSpx);
-    InfeasibleProblemReport produceReport();
+    SingleAnalysis(std::shared_ptr<operations_research::MPSolver> problem_);
+    virtual void run() = 0;
```

Comment:
Not a must-have, but I would advise a more functional style, this interface does not need to be stateful (does not need to manage data members):
```cpp
virtual bool run(operations_research::MPSolve& problem) = 0;
// no data member
```

Reply:
Instead of the boolean, it could actually return a richer "report part" object, to be concatenated in the final report.

Reply:
- About the **problem** as an argument of the **run(...)** method  : 
  In the current state of the code, we have 3 concrete classes : an analyzer (which supervises the whole analysis), and the 2 different analyses. Each analysis owns the problem, but the analyzer does not see the problem at all : it just knows if an analysis failed to find the reason of unfeasibility or succeeded, and if it must stop the analyse there or run another one. 
Should the analyzer see the problem ? At least it's a dependency it does not have.
So in this context, passing the problem to an analysis run function is impossible.
- About the returned report :

Reply:
Done.